docs/mutations.md

```diff
@@ -0,0 +1,56 @@
+# Mutate object after instantion
+
+The _Container_ supports objects to be mutated after it is constructed.
+
+After the _Container_ constructs a new instance of an object, you can specify which other objects will mutate the original object before locking it.
+
+Say we have classes like the following:
+
+```php
+namespace Vendor\Package;
+
+use Aura\Di\Injection\MutationInterface;
+
+class Example
+{
+    protected $foo;
+
+    public function setFoo($foo)
+    {
+        $this->foo = $foo;
+    }
+}
+
+class ExampleMutation implements MutationInterface
+{
+    public function mutate(object $object): object
```

Comment:
I noticed the `MutationInterface` is having `__invoke` and not `mutate`. Also I think there need a better example. Basically in the examples you mentioned about arguments can be passed. But in the class they are missing which makes things a bit hard to digest. I am still trying to figure out the mutation and how they are working. I will look with a fresh eye later.

Reply:
Aight, I will add a better example.